### PR TITLE
[DSM] Feature branch

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/pages/list/List.tsx [hunk 1]
```diff
@@ -107,6 +108,7 @@ const List = () => {
       </PageHeader>
 
       <PageContent>
+        <Dummy size={40} onClick={() => console.log('coucou')}></Dummy>
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/pages/list/List.tsx:111]: To remove before merging on master

**Review by ValentinMumble — APPROVED** [review]

File: .github/actions/extract/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+Repository pandoc-test-output: contains only the generated files from the first Git Action. Pushed here with github-action-push-to-another-repository
+
+And pandoc-test-output can have Git Pages to give access to the files (or just links to the raw version of the files)
+
+## Inputs
+### `source-directory` (argument)
+From the repository that this Git Action is executed the directory that contains the files to be pushed into the repository.
+
+### `destination-github-username` (argument)
+For the repository `https://github.com/cpina/push-to-another-repository-output` is `cpina`. It's also used for the `Author:` in the generated git messages.
```
Comment (tseho) [inline .github/actions/extract/README.md:17]: `cpina/push-to-another-repository-output` ?

File: .github/workflows/dsm-extract.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        steps:
+            -   name: Checkout the current branch
+                uses: actions/checkout@v2.3.2
+            -   name: Download the lib artifact
+                uses: actions/download-artifact@v1
+                with:
+                    name: Lib
+                    path: akeneo-design-system/lib
+            -   name: Bump package version
+                run: rm akeneo-design-system/package.json
```
Comment (tseho) [inline .github/workflows/dsm-extract.yml:65]: bump / rm ?

File: akeneo-design-system/.github/workflows/bump-version.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+
```
Comment (tseho) [inline akeneo-design-system/.github/workflows/bump-version.js:1]: You could talk about this script, I'm not confident in autotagging the releases.
Or maybe if we do for patches, something like a timestamp.

File: akeneo-design-system/.storybook/preview.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import {addDecorator, addParameters} from '@storybook/react';
+import { withThemesProvider } from "themeprovider-storybook";
```
Comment (tseho) [inline akeneo-design-system/.storybook/preview.tsx:3]: is prettier working ?

File: akeneo-design-system/jest.visual.config.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+const baseConfig = require('./jest.unit.config');
+
+module.exports = {
+  ...baseConfig,
+  testMatch: ['**/?(*.)+(visual).ts?(x)'],
```
Comment (tseho) [inline akeneo-design-system/jest.visual.config.js:5]: if there is only one `all.visual.ts` file, should we make it explicit ?

File: akeneo-design-system/package.json [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    "test:visual:run": "jest --config ./jest.visual.config.js",
+    "test:unit:watch": "jest --watch",
+    "eslint": "eslint --config .eslintrc.json --quiet src/**/*.tsx",
+    "lint:fix": "prettier --config .prettierrc.json --parser typescript --write \"./src/**/*.{ts,tsx}\";",
+    "lint:check": "prettier --config .prettierrc.json --parser typescript --check \"./src/**/*.{ts,tsx}\" && eslint --config .eslintrc.json --quiet src/**/*.{ts,tsx}",
+    "storybook:start": "start-storybook -p 6006",
+    "storybook:ci": "start-storybook -p 6006 --ci",
+    "storybook:build": "build-storybook",
+    "lib:build:prod": "yarn prebuild && export NODE_ENV=production && rollup -c",
+    "lib:build:watch": "yarn prebuild-watch && export NODE_ENV=dev && rollup --watch -c",
```
Comment (tseho) [inline akeneo-design-system/package.json:24]: why are the env variables defined after the prebuild command ?

File: akeneo-design-system/rollup.config.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
+// import typescript from 'rollup-plugin-typescript2';
```
Comment (tseho) [inline akeneo-design-system/rollup.config.js:1]: are we sure about keeping rollup ? Webpack, that we already know, can already do the same task.

File: akeneo-design-system/src/components/Checkbox/Checkbox.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+<ArgsTable story="Standard" />
+
+## Variation on state
+<Story name="State">
+  {(args) => {
+    return (
+      <>
+        <Checkbox  {...args} checked={true} label={'Checkbox checked'}/>
+        <Checkbox  {...args} checked={false} undetermined={true} label={'Checkbox undetermined'}/>
```
Comment (tseho) [inline akeneo-design-system/src/components/Checkbox/Checkbox.stories.mdx:62]: IMHO, `undetermined` is confusing.

File: akeneo-design-system/src/shared/test-util.tsx [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import React, {ComponentType, ReactElement, ReactNode} from 'react';
+import {render, RenderOptions} from '@testing-library/react';
+import {ThemeProvider} from 'styled-components';
+import {pimTheme} from '../theme/pim';
+
+const AllTheProviders = ({children}: {children: ReactNode}) => {
+  return <ThemeProvider theme={pimTheme}>{children}</ThemeProvider>;
+};
```
Comment (tseho) [inline akeneo-design-system/src/shared/test-util.tsx:8]: I find the name of this function confusing. It does not really explain what it does.

File: akeneo-design-system/src/shared/test-util.tsx [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+
+const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) =>
+  render(ui, {wrapper: AllTheProviders as ComponentType, ...options});
```
Comment (tseho) [inline akeneo-design-system/src/shared/test-util.tsx:11]: Should we explain why we have a custom render here ?

File: frontend/webpack/requirejs-utils.js [hunk 12]
```diff
@@ -77,6 +77,7 @@ const utils = {
           'require-context': path.resolve(sourceDir, './frontend/webpack/require-context.js'),
           'module-registry': path.resolve(baseDir, './public/js/module-registry.js'),
           routes: path.resolve(baseDir, './public/js/routes.js'),
+          'akeneo-design-system': path.resolve(baseDir, './akeneo-design-system/lib/index.js'),
```
Comment (tseho) [inline frontend/webpack/requirejs-utils.js:80]: It's easier to work with it like this ... but is it really necessary ? Can we have unforeseen consequences if this overwrite the package from `node_modules` ?

File: package.json [hunk 13]
```diff
@@ -25,7 +25,8 @@
     "less": "node $npm_package_config_styles",
     "less-watch": "SHELL=/bin/bash chokidar --verbose \"{$npm_package_config_source/src,src,public/bundles}/**/*.{less,css}\" -c \"yarn run less --dev\"",
     "update-extensions": "node $npm_package_config_source/frontend/build/update-extensions.js",
-    "generate-models": "yarn --cwd=node_modules/@akeneo-pim-community/communication-channel generate-models"
+    "generate-models": "yarn --cwd=node_modules/@akeneo-pim-community/communication-channel generate-models",
+    "watch-front": "concurrently \"yarn webpack-watch\" \"yarn --cwd=akeneo-design-system lib:build:watch\""
```
Comment (tseho) [inline package.json:29]: is this documented ?

File: .github/actions/extract/README.md (hunk 2 above)
Comment (SamirBoulil) [reply .github/actions/extract/README.md:17]: Will fix

File: .github/workflows/dsm-extract.yml [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+        runs-on: ubuntu-latest
+        steps:
+            -   name: Checkout the current branch
+                uses: actions/checkout@v2.3.2
+            -   name: Download the lib artifact
+                uses: actions/download-artifact@v1
+                with:
+                    name: Lib
+                    path: akeneo-design-system/lib
+            -   name: Bump package version
```
Comment (SamirBoulil) [inline .github/workflows/dsm-extract.yml:64]:
```suggestion
            -   name: Remove package version
```

File: akeneo-design-system/.github/workflows/bump-version.js (hunk 4 above)
Comment (SamirBoulil) [reply akeneo-design-system/.github/workflows/bump-version.js:1]: Yes, Ticket created here https://akeneo.atlassian.net/browse/RAC-253

File: akeneo-design-system/.storybook/preview.tsx (hunk 5 above)
Comment (SamirBoulil) [reply akeneo-design-system/.storybook/preview.tsx:3]: Fixed in the Button PRs

File: akeneo-design-system/jest.visual.config.js (hunk 6 above)
Comment (SamirBoulil) [reply akeneo-design-system/jest.visual.config.js:5]: Quentin: We shouldn't forget to add explicit documentation to write visual testing

Comment (SamirBoulil) [reply akeneo-design-system/jest.visual.config.js:5]: And yes, we should be able to add explicit visual tests

File: akeneo-design-system/src/components/Checkbox/Checkbox.stories.mdx (hunk 9 above)
Comment (SamirBoulil) [reply akeneo-design-system/src/components/Checkbox/Checkbox.stories.mdx:62]: When we read the Checkbox on notion https://www.notion.so/akeneo/Checkboxes-a024215ff08340389badc64839e36429. It should have an enum with 3 states

File: akeneo-design-system/src/shared/test-util.tsx (hunk 10 above)
Comment (SamirBoulil) [reply akeneo-design-system/src/shared/test-util.tsx:8]: Let's call it Wrapper

File: frontend/webpack/requirejs-utils.js (hunk 12 above)
Comment (SamirBoulil) [reply frontend/webpack/requirejs-utils.js:80]: It will be imported in node_modules later on.

File: package.json (hunk 13 above)
Comment (SamirBoulil) [reply package.json:29]: Let's rename this command to "dsm-watch" so it reflects the usecase of contributing to the DSM

File: akeneo-design-system/src/components/Checkbox/Checkbox.stories.mdx [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+      <>
+        <Checkbox {...args} readOnly={true} checked={true} label={'Checked disabled'}/>
+        <Checkbox {...args} readOnly={true} checked={false} undetermined={true} label={'Undetermined disabled'}/>
+        <Checkbox {...args} readOnly={true} checked={false} label={'Unchecked disabled'}/>
+      </>
+    )
+  }}
+</Story>
+
+## Animation
```
Comment (ValentinMumble) [inline akeneo-design-system/src/components/Checkbox/Checkbox.stories.mdx:82]: This Story is not needed anymore as we can interact with the playground?

File: .github/actions/extract/README.md (hunk 2 above)
Comment (tseho) [reply .github/actions/extract/README.md:20]: up ?

**Review by StevenVAIDIE — APPROVED** [review]

File: akeneo-design-system/jest.visual.config.js (hunk 6 above)
Comment (ValentinMumble) [reply akeneo-design-system/jest.visual.config.js:5]: Card created: https://akeneo.atlassian.net/browse/RAC-308

**Pull request merged**